pkg/util/util.go

```diff
@@ -29,6 +29,17 @@ func Template(tmpl string, f template.FuncMap, cs *acsapi.OpenShiftManagedCluste
 			replacer := strings.NewReplacer("$", "\\$")
 			return replacer.Replace(b)
 		},
+		"agentPoolIndex": func(r string) int {
+			AgentPoolProfileRoleDummy := acsapi.AgentPoolProfileRole(r)
+			for i, ap := range cs.Properties.AgentPoolProfiles {
+				if ap.Role == AgentPoolProfileRoleDummy {
+					return i
+				}
+			}
+			// This should never happen
+			panic("template failed")
+			return 0
+		},
```

Comment:
I really really dont like how I did this part. I need to sleep on it to think something else, if @pweil- or @jim-minter have any suggestions, let me know. Either change API or logic to inject those values into the template 

Tests will follow when we settle down on implementation and API.